src/lib.rs

```diff
@@ -277,93 +274,77 @@ async fn load_macaroon(
 /// If you have a motivating use case for use of direct data feel free to open an issue and
 /// explain.
 #[cfg_attr(feature = "tracing", tracing::instrument(name = "Connecting to LND"))]
-pub async fn connect<A, CP, MP>(
-    address: A,
+pub async fn connect<CP, MP>(
+    address: String,
     cert_file: CP,
     macaroon_file: MP,
 ) -> Result<Client, ConnectError>
 where
-    A: TryInto<tonic::transport::Endpoint> + std::fmt::Debug + ToString,
-    <A as TryInto<tonic::transport::Endpoint>>::Error: std::error::Error + Send + Sync + 'static,
     CP: AsRef<Path> + Into<PathBuf> + std::fmt::Debug,
     MP: AsRef<Path> + Into<PathBuf> + std::fmt::Debug,
 {
-    let address_str = address.to_string();
-
-    let conn = try_map_err!(address.try_into(), |error| {
-        InternalConnectError::InvalidAddress {
-            address: address_str.clone(),
-            error: Box::new(error),
-        }
-    })
-    .tls_config(tls::config(cert_file).await?)
-    .map_err(InternalConnectError::TlsConfig)?
-    .connect()
-    .await
-    .map_err(|error| InternalConnectError::Connect {
-        address: address_str,
-        error,
-    })?;
+    let config = tls::config(cert_file).await?;
+    let connector = hyper_rustls::HttpsConnectorBuilder::new()
+        .with_tls_config(config.clone())
+        .https_or_http()
+        .enable_http2()
+        .build();
+    let hc = hyper::Client::builder().build(connector);
 
     let macaroon = load_macaroon(macaroon_file).await?;
-
     let interceptor = MacaroonInterceptor { macaroon };
 
+    let svc = InterceptedService::new(hc, interceptor);
+    let uri =
+        Uri::from_str(address.as_str()).map_err(|error| InternalConnectError::InvalidAddress {
+            address,
+            error: Box::new(error),
+        })?;
+
```

Comment:
I think we can inline a bunch of stuff here given it is just used once:

```suggestion
    let connector = hyper_rustls::HttpsConnectorBuilder::new()
        .with_tls_config(tls::config(cert_file).await?)
        .https_or_http()
        .enable_http2()
        .build();
    let macaroon = load_macaroon(macaroon_file).await?;
    
    let svc = InterceptedService::new(hyper::Client::builder().build(connector), MacaroonInterceptor { macaroon });
    let uri =
        Uri::from_str(address.as_str()).map_err(|error| InternalConnectError::InvalidAddress {
            address,
            error: Box::new(error),
        })?;
```

Reply:
applied